### PR TITLE
Improvements to handling of errors and additional tests

### PR DESCRIPTION
**Pull request by connoravo**
- Retries on S3 connections - Improve resiliency to networking issues
- Additional test cases for prefix nuances where they don't match RBAC CSV

Signed-off-by: Connor Avery <ConnorAvery@digital.uc.dwp.gov.uk>

### CODE REVIEW

File: test_s3_tagger.py [hunk 1]
```diff
@@ -382,32 +451,6 @@ def test_tag_path_for_partition(objects_to_tag_partitioned, csv_data):
     assert response2["TagSet"][2]["Value"] == "true", "Object was not tagged correctly"
 
 
-@mock_s3
-def test_tag_path_for_partition(objects_to_tag_partitioned, csv_data):
```
Comment (IanUtley) [inline test_s3_tagger.py:386]: Why have you removed this test?

File: test_s3_tagger.py [hunk 2]
```diff
@@ -464,43 +507,6 @@ def test_tag_folder(csv_data):
     assert response5["TagSet"][2]["Value"] == "true", "Object was not tagged correctly"
 
 
-@mock_s3
-def test_items_that_dont_match_expected_pattern():
```
Comment (IanUtley) [inline test_s3_tagger.py:468]: Why have we removed this test?

File: test_s3_tagger.py (hunk 1 above)
Comment (connoravo) [reply test_s3_tagger.py:386]: Duplicate, see above it

File: test_s3_tagger.py (hunk 2 above)
Comment (connoravo) [reply test_s3_tagger.py:468]: Duplicate / moved up and renamed. See: `test_tag_object_unrecognised_key_for_database_and_table_name_length_two`

**Review by IanUtley — APPROVED** [review]

**Pull request merged**